lightning/src/chain/keysinterface.rs

```diff
@@ -193,19 +195,56 @@ pub struct InMemoryChannelKeys {
 	pub htlc_base_key: SecretKey,
 	/// Commitment seed
 	pub commitment_seed: [u8; 32],
+	/// Local public keys and basepoints
+	pub local_channel_pubkeys: ChannelPublicKeys,
 	/// Remote public keys and base points
 	pub remote_channel_pubkeys: Option<ChannelPublicKeys>,
 	/// The total value of this channel
 	pub channel_value_satoshis: u64,
 }
 
+impl InMemoryChannelKeys {
+	/// Create a new InMemoryChannelKeys
+	pub fn new<C: Signing>(
```

Comment:
If you're gonna have a new, probably best to hide the internal fields, no?

Reply:
Done, but a couple of `pub(crate)` fields due to reference from the enforcing impl